### PR TITLE
[Fix] #37 - Fix build process and run for Carla Agent docker image

### LINKED ISSUE #37: [Bug]: Update Docker Image for Carla Simulator to Fix Build Process
### Current Behavior

https://github.com/una-auxme/paf23/commit/6e8f85ddb0d2479ab9d324d7c7da39b74af97ff4

Description:
We are currently using on outdated versions of the Carla simulator. Our project is using specific commits from the `scenario_runner` and `leaderboard` repositories, which are not compatible with the latest versions of these tools. This problem stems from relying on an outdated Carla simulator version provided by @nylser in his Docker registry.

Details:

Current Approach: The build process uses old commits from `scenario_runner` and `leaderboard` repositories to maintain compatibility with the outdated Carla simulator.
Issue Source: The root of the problem is the outdated version of the Carla simulator in the Docker image provided by @nylser.

Proposed Solution:

Create an Updated Docker Image:
Develop a new Docker image that includes the up-to-date Carla simulator files, as available on the leaderboard website.

Update Repository Branches:
Once the Docker image is updated, align the `scenario_runner` and `leaderboard` repository branches with the latest versions.

Thorough Testing:
Conduct comprehensive tests of the entire setup after updating the Docker image and repository branches to ensure stability and compatibility.

Expected Outcome:
Updating the Docker image and aligning our repository branches with the latest versions should resolve the build issues and improve the overall stability and functionality of our project.

### Expected Behavior

_No response_

### How to reproduce the issue

_No response_

### PR DESCRIPTION
**Pull request by MaxJa4**
# Description

This PR fixes the overall Carla Agent building process and uses the newest leaderboard-2.0 version:
- Updated PythonAPI to the special leaderboard-2.0 v0.9.14-Version (hosted in this repo now as [release](https://github.com/una-auxme/paf23/releases/tag/v0.0.1))
- Updated leaderboard + scenario-runner to latest leaderboard-2.0 version
- Small bugfixes and improvements to Agent-Dockerfile
- Rebuilt Carla Simulator to use the newest leaderboard-2.0 version (located [here](https://github.com/una-auxme/paf23/pkgs/container/paf23/148387355?tag=leaderboard-2.0)) -> also 8GB smaller now
- Fixed linter error messages on `b5 run`
- Fixed ALSA error messages from Unreal-Engine on `b5 run`

--> Building and running works again with the newest updates for leaderboard-2.0!

**Be aware**, that the current implementation (PAF22) is **broken**! The agent doesn't leave its parking spot at the beginning, probably because there's a lot more traffic now.

The rebuilt Carla Simulator Docker image was downloaded from [here](https://leaderboard.carla.org/get_started/#11-download-the-carla-leaderboard-package) and then built with the included Dockerfile.

Fixes #37 #76

Authors: @samuelkuehnel @MaxJa4 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Does this PR introduce a breaking change?

No, it fixes

## Most important changes

Newest leaderboard-2.0 files and working agent docker build & run.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes (might be obsolete with CI later on)

### CODE REVIEW

**Comment by MaxJa4** [comment]
@ll7 @JulianTrommer There is a new CARLA simulator version for [Leaderboard-2.0](https://leaderboard.carla.org/get_started/) available (the one from `ghcr.io/nylser/carla:leaderboard` is from Oct. 22) and needs to be updated along with the PythonAPI (its from the same image and should match). Also saves ~8GB of image space.

For now, I uploaded it to my Docker repo `finalend4/paf23:leaderboard-2.0`, but it would be best to upload it [here](https://github.com/una-auxme/paf23/pkgs/container/paf23) in this repo. I can't do it myself as the repo's GitHub Token is needed.

FYI: The new leaderboard simulator version comes with a few breaking changes, one of which: a **lot** more traffic. Will document that in this PR after everything is done here.

cc @samuelkuehnel

**Comment by ll7** [comment]
I am currently working on pushing the image. I will notify you as soon as the image is online.

File: build/docker/agent/Dockerfile
Comment (ll7) [inline build/docker/agent/Dockerfile:164]: Why did you make this change?

**Comment by ll7** [comment]
The package is now online: https://github.com/una-auxme/paf23/pkgs/container/paf23/148387355?tag=leaderboard-2.0

File: build/docker/agent/Dockerfile
Comment (MaxJa4) [reply build/docker/agent/Dockerfile:164]: Local building failed with xargs without providing an error message, it worked again with pip install.
The need for the xargs variant was due to the git package 'detectron' which needs torch to be installed first (pip install doesn't care about the order of the requirements.txt).
Since we plan to replace the EfficientPS and most Perception things, detectron and other requirements will probably be unused soon anyway. But I can try it again with xargs since it used to work.

Comment (ll7) [reply build/docker/agent/Dockerfile:164]: No, I think your solution is fine. I was mainly curious, and your reasoning wasn't obvious to me.

**Comment by MaxJa4** [comment]
This PR is currently blocked by #76 and also waits for merging of #88.

**Do not merge (yet)**

File: build/docker/agent/Dockerfile_Submission
Comment (MaxJa4) [inline build/docker/agent/Dockerfile_Submission:1]: Was the commit of this file on purpose? Shouldn't this go into a separate PR for issue #77?

**Comment by MaxJa4** [comment]
The non-driving agent is fixed with [603011e](https://github.com/una-auxme/paf23/pull/56/commits/603011e920996c5c9f1723a732b1e6effb1764da).
After #88 is merged, this PR is ready for merging as well.

**Comment by MaxJa4** [comment]
TODO: Rebase to current main & merge

**Review by samuelkuehnel — APPROVED** [review]
Changes work!

**Pull request merged**
